### PR TITLE
fix: use is_serverless so that output is deterministic

### PR DESCRIPTION
**Pull request by huttotw**
## Description

This uses a new variable `is_serverless` so that the resources created by the module are deterministic.

## Checklist before submitting PR for review

- [x] This change requires a doc update, and I've included it
- [x] My code follows the style guidelines of this project
- [x] I have ensured my code is commented and any new terraform variables have proper descriptions

### CODE REVIEW

File: main.tftest.hcl [hunk 1]
```diff
@@ -7,6 +7,7 @@ run "valid_serverless_minimal_details" {
     vpc_id         = "my-vpc"
     workgroup_arn  = "workgroup_arn"
     s3_bucket_name = "my-bucket"
+    is_serverless  = true
```
Comment (camphillips22) [inline main.tftest.hcl:10]: Do we need to add some validation that `workgroup_arn` should be supplied when is_serverless is true?

**Review by camphillips22 — APPROVED** [review]
LGTM!

**Comment by huttotw** [comment]
@camphillips22 looks like we have to bump all the way to [Terraform v1.9.0](https://github.com/hashicorp/terraform/releases/tag/v1.9.0) in order to enable dependent variable validation, I'm hesitant to bump to such a recent version at the expense of compatibility.

**Pull request merged**